NOTES-VALGRIND.md

```diff
@@ -51,7 +51,7 @@ used to control what capabilities OpenSSL uses.
 As of valgrind-3.15.0 on Linux/x86_64, instructions up to AVX2 are
 supported. Setting the following disables instructions beyond AVX2:
 
-`OPENSSL_ia32cap=":0"`
+`OPENSSL_ia32cap=":0:0:0:0"`
```

Comment:
It looks like the original `OPENSSL_ia32cap=":0"` doesn't disable post AVX2 instructions anymore.
This could behave as a breaking change for some setups.

Reply:
It is easy to adjust with a loop here: https://github.com/openssl/openssl/blob/c87ea2e7b6266e2fc2d7b2aecefa276c96a767cf/crypto/cpuid.c#L145 to zeroize the rest of indexes if needed.

If the adjustment of `OPENSSL_ia32cap` environment variable is a very specific action and it suppose to be manual and explicit - I would advise to keep it as, so the users know and understand what they are turning on and off. There is some past experience from the customers that the implicit disable ‘feature’ had teams doing a lot of debugging, to figure out the RSA performance anomaly.  Ultimately it was just the ‘implicit’ zeroing of `OPENSSL_ia32cap` vectors that caused the problem.

On the other hand, if it is common and advisable to use `OPENSSL_ia32cap=":0"` for the debug, maybe it is worth to keep the old behavior as and make it better documented. However, it may lead to some unexpected results on any post AVX2 HW.

@t8m, do you have any thoughts on how it is better to proceed from OpenSSL maintainers experience?

Reply:
IMO we need to keep the backwards compatibility for this.

Reply:
@t8m, I've updated the behavior for the environment variable and documentation, please, check if that is what expected: https://github.com/openssl/openssl/pull/25709/commits/0c81445cc4b108434c18a7c1ffadecaccca3895d
Now it should not break the expected behavior for cases as OPENSSL_ia32cap=":0" - all the following indexes will be set to 0. It also works for example like:
OPENSSL_ia32cap="::0" - zeroize starting 3rd value or OPENSSL_ia32cap=":0::::" - zeroize only the second one and keep the remaining untouched.

My only concern now is for the situation if OPENSSL_ia32cap would be expanded again, then OPENSSL_ia32cap=":0::::" will zeroize the new indexes after existing once, however, this looks like a really exotic case to turn off all AVX2 and leave the rest unchanged.